src/runtime_src/core/common/sensor.cpp

```diff
@@ -16,8 +13,248 @@
  * License for the specific language governing permissions and limitations
  * under the License.
  */
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+#define XRT_CORE_COMMON_SOURCE
+
+// Local - Include Files
 #include "sensor.h"
-#include <boost/property_tree/json_parser.hpp>
+#include "query_requests.h"
+#include "utils.h"
+
+// System - Include Files
+#include <sstream>
+#include <iomanip>
+
+// ------ S T A T I C   F U N C T I O N S -------------------------------------
+namespace {
+/**
+ * Saves voltage-current pair of a sensor into a boost::property_tree
+ * Converts mV and mA into V and A before adding to the tree
+ * 
+ * @tparam QRVoltage voltage query for a sensor; query::noop if query DNE
+ * @tparam QRCurrent current query for the same sensor; query::noop if query DNE
+ * @param loc_id human readable sensor identifier
+ * @desc description about the sensor
+ */
+template <typename QRVoltage, typename QRCurrent>
+static boost::property_tree::ptree
+populate_sensor(const xrt_core::device * device, const std::string& loc_id, const std::string& desc)
+{
+  boost::property_tree::ptree pt;
+  pt.put("id", loc_id);
+  pt.put("description", desc);
+
+  uint64_t voltage = 0, current = 0;
```

Comment:
Please uniquely quality each variable.
```suggestion
  uint64_t voltage = 0;
  unit64_t current = 0;
```